### PR TITLE
Beacon Color (L4D1/L4D2)

### PR DESCRIPTION
**Pull request by MartLegion**
Added inverse color for L4D1/L4D2 game engine,

Since beacon is based on the team color chat. 

On L4D1/L4D2:
*Survivors are from Team 2 and output to chat in Blue chat color, but the beacon is Red.
*Infecteds are from Team 3 and output to chat in Red chat color, but the beacon is Blue.

Also:
*Modified "if else" to "switch case". 
*Added orange color to the hidden team on L4D2 (Team 4).
*Updated unknown teams to white color.
*Maintained green color to Spec players. (didn't change).

### CODE REVIEW

**Comment by asherkin** [comment]
The red/blue isn't based on any concept of "good guys" and "bad guys", it is the team colours from team chat.

**Comment by MartLegion** [comment]
Thanks for the info, I gonna fix the description.
But, anyway, in L4D1/L4D2 its working like this way (without the PR).
Blue color chat is outputing a Red beacon.
Red color chat is outputing a Blue beacon.

File: plugins/funcommands/beacon.sp [hunk 1]
```diff
@@ -86,30 +86,50 @@ public Action Timer_Beacon(Handle timer, any value)
...
 	GetClientAbsOrigin(client, vec);
 	vec[2] += 10;
 	
+	int teamColor[4];
+	
 	if (g_BeamSprite > -1 && g_HaloSprite > -1)
 	{
-		TE_SetupBeamRingPoint(vec, 10.0, g_Cvar_BeaconRadius.FloatValue, g_BeamSprite, g_HaloSprite, 0, 15, 0.5, 5.0, 0.0, greyColor, 10, 0);
-		TE_SendToAll();
+		EngineVersion engine = GetEngineVersion();
```
Comment (KyleSanderson) [inline plugins/funcommands/beacon.sp]: static EngineVersion engine

Comment (MartLegion) [reply plugins/funcommands/beacon.sp]: Updated.

**Comment by MartLegion** [comment]
I made some changes but your "change request" doesn't compile. Please review again.

**Pull request closed without merging**